src/org/rascalmpl/uri/jar/JarFileResolver.java

```diff
@@ -84,11 +84,7 @@ public String[] list(ISourceLocation jar, String path) throws IOException {
         if (!path.endsWith("/") && !path.isEmpty()) {
             path = path + "/";
         }
-        try {
-            return getFileHierchyCache(jar).directChildren(path);
-        }
-        catch (IOException e) {
-            return new String[0];
-        }
+        
+        return getFileHierchyCache(jar).directChildren(path);
```

Comment:
Was this big only there for the jar resolver?

Reply:
It's possible that some of the Eclipse resolvers had the same issue, but the current (short) list of schemes does not seem to have this bug.

Reply:
I disagree.

In `FileURIResolver` we use [`File::list()`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/File.html#list()) and here it also says:

> An array of strings naming the files and directories in the directory denoted by this abstract pathname. The array will be empty if the directory is empty. Returns null if this abstract pathname does not denote a directory, or if an I/O error occurs.

So it will not throw an `IOException` either.

I don't think this change should be added to this PR, as there are quite a few places that might depend on this behavior change.